sdk/core/az_core/inc/az_credentials.h

```diff
@@ -73,8 +73,11 @@ typedef struct
  * @brief function callback definition as a contract to be implemented for a credential
  *
  */
-typedef AZ_NODISCARD az_result (
-    *_az_credential_apply_fn)(void* credential_options, _az_http_request* ref_request);
+typedef AZ_NODISCARD az_result (*_az_credential_apply_policy_fn)(
```

Comment:
Delete this typedef entirely and use _az_http_policy_process_fn instead. You'll have to move that typedef here.

Reply:
Done